docs/rapha-rover/specification/ros-api.mdx

```diff
@@ -0,0 +1,567 @@
+---
+title: ROS API
+sidebar_label: ROS API
+---
+
+:::warning
+
+The ROS API in the beta version is subject to changes.
+
+:::
+
+## Subscribed topics
+
+- `controller/cmd_ackermann` ([ackermann_msgs/msg/AckermannDrive])
+
+  Steers the robot when operating in the Ackermann steering mode.
+
+- `controller/cmd_turn_in_place` (?) **Not implemented yet**
+
+  Steers the robot when operating in the Turn-In-Place steering mode.
+
+- `controller/cmd_vel` ([geometry_msgs/msg/Twist]) **Not implemented yet**
+
+  Steers the robot using `Twist` commands. More standardized way to control the
+  robot. \
+  Useful for some navigation stacks. Not recommended for manual teleoperation.
+
+- `controller/led_strip_state` ([rapha_interfaces/msg/LedStripState])
+
+  Sets a new user state for all the LEDs in the LED strip.
+
+- `controller/led_panel_state` ([rapha_interfaces/msg/LedPanelState]) **Not
+  implemented yet**
+
+  Sets a new user state for all the LEDs in the specified LED panel.
+
+- `system/shutdown` ([std_msgs/msg/Empty])
+
+  Performs system shutdown.
+
+- `system/reboot` ([std_msgs/msg/Empty])
+
+  Performs system reboot.
+
+## Published Topics
+
+- `controller/imu` ([sensor_msgs/msg/Imu])
+
+  Current gyroscope and accelerometer readings from the onboard IMU.
+
+- `controller/odom` ([nav_msgs/msg/Odometry])
+
+  Odometry calculated from wheel encoders and the onboard IMU.
+
+- `controller/steering_mode` ([rapha_interfaces/msg/SteeringMode]) **Not
+  implemented yet**
+
+  Current steering mode. Published whenever the mode changes. \
+  Uses transient local durability to deliver last message to joining
+  subscriptions.
+
+- `controller/diagnostics/imu` ([rapha_interfaces/msg/ImuDiagnostics])
+
+  Diagnostic information about the IMU module. \
+  Includes module temperature and calibration data.
+
+- `controller/diagnostics/motors` ([rapha_interfaces/msg/MotorDiagnostics])
+
+  Diagnostic information about the motors. \
+  Includes motor temperatures, power consumption and fault data.
+
+- `controller/power_system_state` ([rapha_interfaces/msg/PowerSystemState])
+
+  Current information about the power system.
+
+- `oak/imu/data` ([sensor_msgs/msg/Imu])
+
+  Current gyroscope and accelerometer readings from the Oak IMU.
+
+- `oak/left/camera_info` ([sensor_msgs/msg/CameraInfo])
+
+  Calibration data for the left Oak camera.
+
+- `oak/left/image_raw` ([sensor_msgs/msg/Image])
+
+  Raw images from the left Oak camera.
+
+- `oak/rgb/camera_info` ([sensor_msgs/msg/CameraInfo])
+
+  Calibration data for the center (RGB) Oak camera.
+
+- `oak/rgb/image_raw` ([sensor_msgs/msg/Image])
+
+  Raw images from the center (RGB) Oak camera.
+
+- `oak/right/camera_info` ([sensor_msgs/msg/CameraInfo])
+
+  Calibration data for the right Oak camera.
+
+- `oak/right/image_raw` ([sensor_msgs/msg/Image])
+
+  Raw images from the right Oak camera.
+
+- `oak/stereo/camera_info` ([sensor_msgs/msg/CameraInfo])
+
+  Calibration data for **TODO**
```

Comment:
Is ` **TODO** ` left on purpose?

Reply:
I forgot about it. Still not sure how to name it. `Calibration data for the depth Oak camera.` is one option but not really accurate as there is not physical camera there.

Reply:
How about now?